`ModSimplifierAILEngine::_handle_binop_Sub` reaches the unreachable

### Description

```
Traceback (most recent call last):
  File "/megatest/analyze_binary.py", line 157, in _step_context
    yield
  File "/megatest/analyze_binary.py", line 223, in _execute_job_steps
    dec = proj.analyses.Decompiler(func, cfg=cfg, preset=preset, fail_fast=True)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/megatest/venv/lib/python3.12/site-packages/angr/analyses/analysis.py", line 269, in __call__
    r = w(*args, **kwargs)
        ^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/contextlib.py", line 81, in inner
    return func(*args, **kwds)
           ^^^^^^^^^^^^^^^^^^^
  File "/megatest/venv/lib/python3.12/site-packages/angr/analyses/analysis.py", line 254, in wrapper
    oself.__init__(*args, **kwargs)
  File "/megatest/venv/lib/python3.12/site-packages/angr/analyses/decompiler/decompiler.py", line 143, in __init__
    self._decompile()
  File "/megatest/venv/lib/python3.12/site-packages/angr/utils/timing.py", line 55, in timed_func
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/megatest/venv/lib/python3.12/site-packages/angr/analyses/decompiler/decompiler.py", line 235, in _decompile
    clinic = self.project.analyses.Clinic(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/megatest/venv/lib/python3.12/site-packages/angr/analyses/analysis.py", line 269, in __call__
    r = w(*args, **kwargs)
        ^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/contextlib.py", line 81, in inner
    return func(*args, **kwds)
           ^^^^^^^^^^^^^^^^^^^
  File "/megatest/venv/lib/python3.12/site-packages/angr/analyses/analysis.py", line 254, in wrapper
    oself.__init__(*args, **kwargs)
  File "/megatest/venv/lib/python3.12/site-packages/angr/analyses/decompiler/clinic.py", line 184, in __init__
    self._analyze_for_decompiling()
  File "/megatest/venv/lib/python3.12/site-packages/angr/analyses/decompiler/clinic.py", line 234, in _analyze_for_decompiling
    ail_graph = self._decompilation_simplifications(ail_graph)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/megatest/venv/lib/python3.12/site-packages/angr/analyses/decompiler/clinic.py", line 563, in _decompilation_simplifications
    ail_graph = self._run_simplification_passes(ail_graph, stage=OptimizationPassStage.AFTER_GLOBAL_SIMPLIFICATION)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/megatest/venv/lib/python3.12/site-packages/angr/utils/timing.py", line 55, in timed_func
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/megatest/venv/lib/python3.12/site-packages/angr/analyses/decompiler/clinic.py", line 1274, in _run_simplification_passes
    a = pass_(
        ^^^^^^
  File "/megatest/venv/lib/python3.12/site-packages/angr/utils/timing.py", line 55, in timed_func
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/megatest/venv/lib/python3.12/site-packages/angr/analyses/decompiler/optimization_passes/mod_simplifier.py", line 77, in __init__
    self.analyze()
  File "/megatest/venv/lib/python3.12/site-packages/angr/analyses/decompiler/optimization_passes/optimization_pass.py", line 87, in analyze
    self._analyze(cache=cache)
  File "/megatest/venv/lib/python3.12/site-packages/angr/analyses/decompiler/optimization_passes/mod_simplifier.py", line 90, in _analyze
    new_block = self.engine.process(state=self.state.copy(), block=old_block.copy())
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/megatest/venv/lib/python3.12/site-packages/angr/engines/light/engine.py", line 70, in process
    return self._process(state, block=block, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/megatest/venv/lib/python3.12/site-packages/angr/engines/light/engine.py", line 654, in _process
    stmt_data = self._process_stmts(whitelist=whitelist)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/megatest/venv/lib/python3.12/site-packages/angr/engines/light/engine.py", line 694, in _process_stmts
    result.append(self._stmt(stmt))
                  ^^^^^^^^^^^^^^^^
  File "/megatest/venv/lib/python3.12/site-packages/angr/engines/light/engine.py", line 700, in _stmt
    return self._stmt_handlers[stmt_type_name](stmt)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/megatest/venv/lib/python3.12/site-packages/angr/analyses/decompiler/optimization_passes/engine_base.py", line 69, in _handle_stmt_Assignment
    src = self._expr(stmt.src)
          ^^^^^^^^^^^^^^^^^^^^
  File "/megatest/venv/lib/python3.12/site-packages/angr/engines/light/engine.py", line 732, in _expr
    return self._expr_handlers[expr_type_name](expr)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/megatest/venv/lib/python3.12/site-packages/angr/engines/light/engine.py", line 756, in _handle_expr_BinaryOp
    return self._binop_handlers[expr.op](expr)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/megatest/venv/lib/python3.12/site-packages/angr/analyses/decompiler/optimization_passes/mod_simplifier.py", line 44, in _handle_binop_Sub
    assert False, "Unreachable"
           ^^^^^
AssertionError: Unreachable
```

### Steps to reproduce the bug

decompile_full	[b79e998ccc9d0d6d431645c87c7802ae90fe1a2522bd77eb16cdbf65f6f88507](https://angr.megatest.cc/binary/by-sha256/b79e998ccc9d0d6d431645c87c7802ae90fe1a2522bd77eb16cdbf65f6f88507/content)	0x140013270

### Environment

angr @ 250cca783f50d5134ad743aba4748aae17d70327

### Additional context

_No response_

--- a/angr/analyses/decompiler/optimization_passes/mod_simplifier.py
+++ b/angr/analyses/decompiler/optimization_passes/mod_simplifier.py
@@ -40,10 +40,16 @@ class ModSimplifierAILEngine(SimplifierAILEngine):
                 x_1 = operand_0
                 c_0 = operand_1.operands[1]
                 c_1 = operand_1.operands[0].operand.operands[1]
-            else:
-                assert False, "Unreachable"
 
-            if x_0 is not None and x_1 is not None and x_0.likes(x_1) and c_0.value == c_1.value:
+            # pylint: disable=too-many-boolean-expressions
+            if (
+                x_0 is not None
+                and x_1 is not None
+                and c_0 is not None
+                and c_1 is not None
+                and x_0.likes(x_1)
+                and c_0.value == c_1.value
+            ):
                 return Expr.BinaryOp(expr.idx, "Mod", [x_0, c_0], expr.signed, **expr.tags)
 
         if (operand_0, operand_1) != (expr.operands[0], expr.operands[1]):

